src/commands/base.ts

```diff
@@ -251,6 +251,28 @@ export default abstract class Base extends Command {
         return params
     }
 
+    public async populateParametersWithZod
+    <ResourceType extends Record<string, ZodTypeAny>>(
+        schema: ZodObject<ResourceType>, 
+        prompts: Prompt[],
+        flags: Record<string, unknown>,
+    ): Promise<z.infer<typeof schema> | undefined> {
+        let input = flags
+        if (!flags.headless) {
+            const filteredPrompts = filterPrompts(prompts, flags)
+            const answers = await this.populateParametersWithInquirer(filteredPrompts)
+            input = mergeFlagsAndAnswers(flags, answers)
+        }
+
+        const parse = schema.safeParse(input)
+
+        if (!parse.success) {
+            this.writer.showError(parse.error.message)
```

Comment:
I think we should throw an error here, otherwise it's harder for functions calling this method to determine if there was an error or not

Reply:
good point, i changed it so `populateParametersWithZod` will throw and then the actual run methods for the commands can catch and handle the error